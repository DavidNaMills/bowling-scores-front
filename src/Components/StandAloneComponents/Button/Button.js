import React from 'react';
import classes from './Button.module.scss';



const Button = ({ click = () => { }, label, isDisabled = false, type = 'default' }) => {
    let style = [classes.button];

    if (isDisabled) {
        style = style.concat(classes.button__disabled);
    } else {

        switch (type.toLowerCase()) {
            case 'warning':
                style = style.concat(classes.button__warning);
                break;
            case 'danger':
                style = style.concat(classes.button__danger);
                break;
            default:
                style = style.concat(classes.button__default);
                break;
        }
    }

    return (
        <button onClick={click} disabled={isDisabled} className={style.join(' ')}>{label}</button>
    );
};

export default Button;