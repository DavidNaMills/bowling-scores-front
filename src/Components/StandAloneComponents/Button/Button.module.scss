@import '../../../styles/colors.scss';


.button{
    border: none;
    font-size: 16px;
    padding: 10px;
    min-width: 100px;
    font-weight: 550;

    &__disabled{
        background-color: $btn_disabled;
    }
    
    &__isFull{
        width: 100%;
    }

    &__default{
        background-color: $btn_default_bkg;
        color: $btn_font_clr_light;
        
        &:hover{
            background-color: $btn_default_bkg_active_hover;
        }
        &:active{
            background-color: $btn_default_bkg_active;
        }
    }

    &__warning{
        background-color: $btn_warning_bkg;
        color: $btn_font_clr;
        
        &:hover{
            background-color: $btn_warning_bkg_active_hover;
        }
        &:active{
            background-color: $btn_warning_bkg_active;
        }    
    }
    
    &__danger{
        background-color: $btn_danger_bkg;
        color: $btn_font_clr_light;
        
        &:hover{
            background-color: $btn_danger_bkg_active_hover;
        }
        &:active{
            background-color: $btn_danger_bkg_active;
        }    
    }

    &__lightGreen{
        background-color: $btn_lightGreen_bkg;
        color: $btn_font_clr;
        
        &:hover{
            background-color: $btn_lightGreen_bkg_active;
        }
        &:active{
            background-color: $btn_lightGreen_bkg_active_hover;
        }    
    }

    &__darkGreen{
        background-color: $btn_darkGreen_bkg;
        color: $btn_font_clr_light;
        
        &:hover{
            background-color: $btn_darkGreen_bkg_active;
        }
        &:active{
            background-color: $btn_darkGreen_bkg_active_hover;
        }    
    }

    &__blue{
        background-color: $btn_blue_bkg;
        color: $btn_font_clr_light;
        
        &:hover{
            background-color: $btn_blue_bkg_active;
        }
        &:active{
            background-color: $btn_blue_bkg_active_hover;
        }    
    }

    &__lightBlue{
        background-color: $btn_lightBlue_bkg;
        color: $btn_font_clr;
        
        &:hover{
            background-color: $btn_lightBlue_bkg_active;
        }
        &:active{
            background-color: $btn_lightBlue_bkg_active_hover;
        }    
    }

    &__darkBlue{
        background-color: $btn_darkBlue_bkg;
        color: $btn_font_clr_light;
        
        &:hover{
            background-color: $btn_darkBlue_bkg_active;
        }
        &:active{
            background-color: $btn_darkBlue_bkg_active_hover;
        }    
    }

    &__lightRed{
        background-color: $btn_lightRed_bkg;
        color: $btn_font_clr;
        
        &:hover{
            color: $btn_font_clr_light;
            background-color: $btn_lightRed_bkg_active;
        }
        &:active{
            background-color: $btn_lightRed_bkg_active_hover;
        }    
    }

    &__lightYellow{
        background-color: $btn_lightYellow_bkg;
        color: $btn_font_clr;
        
        &:hover{
            color: $btn_font_clr_light;
            background-color: $btn_lightYellow_bkg_active;
        }
        &:active{
            background-color: $btn_lightYellow_bkg_active_hover;
        }    
    }



}