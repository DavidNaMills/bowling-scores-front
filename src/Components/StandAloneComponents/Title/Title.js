import React from 'react';
import classes from './Title.module.scss';

const Title = ({ label = '', ttlType = 'sub' }) => {
    let item;

    switch (ttlType.toLowerCase()) {
        case 'main':
            item = <h1 className={classes.title__main}>{label}</h1>
            break;
        case 'section':
            item = <h2 className={classes.title__section}>{label}</h2>
            break;
        case 'sub':
            item = <h3 className={classes.title__sub}>{label}</h3>
            break;
        default:
            item = <h3 className={classes.title__sub}>{label}</h3>
            break;
    }

    return (
        <div className={classes.title}>
            {item}
        </div>
    )
}

export default Title;