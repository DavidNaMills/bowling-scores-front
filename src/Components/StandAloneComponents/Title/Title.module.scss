@import '../../../styles/colors.scss';
@import '../../../styles/fonts.scss';

.title{
    // -webkit-box-shadow: $title_shadow_full;
    // -moz-box-shadow: $title_shadow_full;
    // box-shadow: $title_shadow_full;
    padding: 2px;
    text-align: center;
    font-family: $header;

    &__main{
        font-size: $header_size_main;
        color: $title_main_color;
    }
    
    &__section{
        font-size: $header_size_section;
        color: $title_rest_color;
    }
    &__sub{
        font-size: $header_size_sub;
        color: $title_rest_color;
    }
}