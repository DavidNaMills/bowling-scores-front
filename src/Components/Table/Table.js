import React from 'react';
import classes from './Table.module.scss';

const Table = ({ caption = '', tableClass = classes.table, data, showRowNum = false, selectRow = () => { }, ...rest }) => (
    <div>
        {caption && <p className={classes.caption}>{caption}</p>}
        <table className={tableClass}>
            <thead>
                {showRowNum && <th></th>}
                {data.headers.map((x, i) =>
                    <th key={i}>
                        {x}
                    </th>
                    , 0)}
            </thead>
            <tbody>
                {
                    data.rows.map((x, i) =>
                        <tr
                            key={i}
                            className={x.style && typeof (x.style.trow) === 'string' ? x.style.trow : ''}
                            style={x.style && typeof (x.style.trow) === 'object' ? x.style.trow : {}}
                            onClick={() => selectRow(x.id ? x.id : i + 1)}
                        >
                            {showRowNum && <td key={`${i}0`}>{i + 1}</td>}
                            {
                                x.values.map((y, o) =>
                                    <td key={`${i}-${o}`}>
                                        {y}
                                    </td>
                                    , 1)
                            }
                        </tr>
                    )
                }
            </tbody>
        </table>
    </div>
)

export default Table;

// const tHeaders = ['name', 'Average', 'Pinfall', 'Rank'];
// const tData = [
//     {
//         id: '',
//         values: ['david', 135, 2445, '+ 2'],
//         style: { tr: classes.testRowClass }  //optional
//     },
//     {
//         id: '5vd5f3d5cvx3cv5xz53xcz',
//         values: ['alan', 189, 2222, '-2'],
//         style: { trow: { color: 'red' } }  //optional
//     }
// ];