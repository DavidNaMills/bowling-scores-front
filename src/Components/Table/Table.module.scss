.table{
    width: 100%;
    border: solid black 1px;
    border-collapse: collapse;
    color: black;
    

    thead {
        
        background: rgb(255,255,255);
        background: -moz-linear-gradient(180deg, rgba(255,255,255,1) 37%, rgba(201,201,201,1) 100%);
        background: -webkit-linear-gradient(180deg, rgba(255,255,255,1) 37%, rgba(201,201,201,1) 100%);
        background: linear-gradient(180deg, rgba(255,255,255,1) 37%, rgba(201,201,201,1) 100%);
        filter: progid:DXImageTransform.Microsoft.gradient(startColorstr="#ffffff",endColorstr="#c9c9c9",GradientType=1);
    }

    th{
        font-size: 16px;
        border: solid rgb(111, 111, 111) 1px;
        color: black;
        padding: 5px;
    }
    
    td{
        font-size: 14px;
        border: solid rgb(111, 111, 111) 1px;
        padding: 5px;
        text-align: center;
    }  
}

.caption{
    font-size: 14px;
    padding:0;
    margin-top:15px;
    margin-bottom:5px;
}