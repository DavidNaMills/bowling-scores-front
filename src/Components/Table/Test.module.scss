.testTable{
    background-color: green;
    color: blue;
}

.testRowClass{
    background-color: yellow;
    color: red;
}