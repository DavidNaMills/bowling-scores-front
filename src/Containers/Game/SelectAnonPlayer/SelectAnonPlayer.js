import React, { useState } from 'react';
import classes from './SelectAnonPlayer.module.scss';
import spacing from '../../../styles/shared/spacing.module.scss';


import Title from '../../../Components/StandAloneComponents/Title/Title';
import Table from '../../../Components/Table/Table';
import Button from '../../../Components/StandAloneComponents/Button/Button';

import pinfallSort from '../../../helpers/pinfallSort/pinfallSort';
import tableParser from '../../../helpers/tableGamesDataModifier/tableGamesDataModifier';
import { playerTableStyle } from '../../../Containers/Game/playerTableStyle';
import useLocalStorage from '../../../Hooks/useLocalStorage/useLocalStorage';

const label1 = 'Please select the name you played as'

const tHeaders = ['Player', 'Average', 'Pinfall'];

const SelectAnonPlayer = (props) => {
    const { removeFromStorage } = useLocalStorage();
    const {
        game,
        cancelClick,
        submitClick
    } = props;

    const [sel, setSel] = useState(null)

    return (
        <div>
            <Title label={label1} />
            <Table
                data={{
                    headers: tHeaders,
                    rows: pinfallSort(tableParser(game))
                }}
                caption='Select Yourself'
                selectRow={setSel}
            />

            {sel &&
                    <p className={spacing.largeExtra} style={{ ...playerTableStyle(game.players[sel].color), padding: '10px', textAlign: 'center' }}>
                        {`Selected Player: ${game.players[sel].name}`}
                    </p>
            }

            <div className={spacing.largeExtra}>
                <Button click={() => submitClick(sel)} label={'Submit'} isDisabled={!sel ? true : false} type='lightgreen' isFull />
            </div>
            <div className={spacing.extra}>
                <Button click={()=>{
                    removeFromStorage();
                    cancelClick();
                }
                } label={'Cancel'} type='lightred' isFull />
            </div>
        </div>
    )
}

export default SelectAnonPlayer;