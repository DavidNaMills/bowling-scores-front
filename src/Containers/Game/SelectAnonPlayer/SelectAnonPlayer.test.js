import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import SelectAnonPlayer from './SelectAnonPlayer';
import Title from '../../../Components/StandAloneComponents/Title/Title';
import Table from '../../../Components/Table/Table';
import Button from '../../../Components/StandAloneComponents/Button/Button';


const tempData = {
    createdAt: "2020-01-08T06:22:59.878Z",
    games: {
        1: { 'debb53c3-ad2a-4093-b89b-2a5025416f6c': { name: "gfdgfds", score: 12 } },
        2: { 'debb53c3-ad2a-4093-b89b-2a5025416f6c': { name: "gfdgfds", score: 24 } },
        3: { 'debb53c3-ad2a-4093-b89b-2a5025416f6c': { name: "gfdgfds", score: 154 } },
    },
    players: {
        'debb53c3-ad2a-4093-b89b-2a5025416f6c': { name: "gfdgfds", color: "17, 127, 14", id: "debb53c3-ad2a-4093-b89b-2a5025416f6c" }
    },
    _id: "123789456123abc"
}

const mockPlayerSelect = jest.fn();

const props = {
    game: tempData,
    playerSelect: mockPlayerSelect
}

describe('<SelectAnonPlayer /> component test suite', () => {
    it('matches the snapshot', () => {
        const comp = renderer.create(<SelectAnonPlayer {...props}/>);
        const app = comp.toJSON();
        expect(app).toMatchSnapshot();
    });


    const wrapper = mount(<SelectAnonPlayer {...props}/>);

    it('exists', () => {
        expect(wrapper.exists()).toBeTruthy();
    });

    it('should contain 2 <Title /> components', () => {
        const ttl = wrapper.find(Title);
        expect(ttl.length).toBe(1);

    });

    it('should contain a <Table /> component', ()=>{
        const tbl = wrapper.find(Table);
        expect(tbl.length).toBe(1);
        expect(wrapper.find('tr').length).toBe(2)
    });

    it('should contain 2 <Button/> components', ()=>{
        const btn = wrapper.find(Button);
        expect(btn.length).toBe(2);
    });
});