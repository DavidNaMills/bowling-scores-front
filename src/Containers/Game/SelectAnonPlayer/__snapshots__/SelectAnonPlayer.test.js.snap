// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectAnonPlayer /> component test suite matches the snapshot 1`] = `
<div>
  <div
    className="title"
  >
    <h3
      className="title__sub"
    >
      Please select the name you played as
    </h3>
  </div>
  <div>
    <p
      className="caption"
    >
      Select Yourself
    </p>
    <table
      className="table"
    >
      <thead>
        <tr>
          <td>
            Player
          </td>
          <td>
            Average
          </td>
          <td>
            Pinfall
          </td>
        </tr>
      </thead>
      <tbody>
        <tr
          className=""
          onClick={[Function]}
          style={
            Object {
              "background": "linear-gradient(180deg, rgba(255,255,255,1) 37%, rgba(17, 127, 14,1) 100%)",
            }
          }
        >
          <td>
            gfdgfds
          </td>
          <td>
            63
          </td>
          <td>
            190
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    className="largeExtra"
  >
    <button
      className="button button__isFull button__disabled"
      disabled={true}
      onClick={[Function]}
    >
      Submit
    </button>
  </div>
  <div
    className="extra"
  >
    <button
      className="button button__isFull button__lightRed"
      disabled={false}
      onClick={[Function]}
    >
      Cancel
    </button>
  </div>
</div>
`;
